packages/graphql/lib/src/core/query_manager.dart

```diff
@@ -269,23 +271,61 @@ class QueryManager {
   }
 
   QueryResult _mapFetchResultToQueryResult(
-    FetchResult fetchResult, {
+    FetchResult fetchResult,
+    BaseOptions options, {
     bool loading,
     bool optimistic = false,
   }) {
     List<GraphQLError> errors;
 
-    if (fetchResult.errors != null) {
+    // check if there are errors and implement error policy
+    // add errors if the [ErrorPolicy] allows to report errors i.e. it errorPolicy != ignore
+    if (_shouldReturnError(fetchResult, options.errorPolicy)) {
       errors = List<GraphQLError>.from(fetchResult.errors.map<GraphQLError>(
         (dynamic rawError) => GraphQLError.fromJSON(rawError),
       ));
     }
 
+    dynamic data;
+
+    // return data is there are no errors
+    if (_shouldReturnData(fetchResult, options.errorPolicy)) {
+      data = fetchResult.data;
+    }
+
     return QueryResult(
-      data: fetchResult.data,
+      data: data,
       errors: errors,
       loading: loading,
       optimistic: optimistic,
     );
   }
+
+  bool _shouldReturnError(FetchResult fetchResult, ErrorPolicy errorPolicy) {
+    // Return errors as long as there are errors and errorPolicy allows it
+    // if errorPolicy is none or all
+    if (fetchResult.errors != null &&
+        (errorPolicy != ErrorPolicy.none || errorPolicy == ErrorPolicy.all)) {
+      return true;
+    }
+
+    return false;
+  }
+
+  bool _shouldReturnData(FetchResult fetchResult, ErrorPolicy errorPolicy) {
+    // if there are no errors, return data
+    if (fetchResult.errors == null) {
+      return true;
+    }
+
+    // if there is data and errors, return data if the error policy allows it
+    // if errorPolicy is return all or ignore errors, return data
+    if (fetchResult.errors != null &&
```

Comment:
for this and in `_shouldReturnErrors` I think a switch statement would be clearer:
```dart
switch (options.errorPolicy) {
  case ErrorPolicy.ignore:
  case ErrorPolicy.all:
    return false;
  case ErrorPolicy.none:
  default:
    return true;
}
```
but that's a pretty minor thing - overall this is great and I'm green lighting it 👍